src/core/instance/state.js

```diff
@@ -142,7 +142,11 @@ function initMethods (vm: Component) {
   const methods = vm.$options.methods
   if (methods) {
     for (const key in methods) {
-      vm[key] = bind(methods[key], vm)
+      if (methods[key] != null) {
+        vm[key] = bind(methods[key], vm)
+      } else {
+        warn(`The method ${key} on vue instance is undefined.`, vm)
```

Comment:
can we add `else if (process.env.NODE_ENV !== 'production')` here so that the warning can be stripped?